### PR TITLE
adding some fixes to cpan skeleton

### PR DESCRIPTION
**Pull request by jerowe**
@msarahan, here is my PR for https://github.com/conda/conda-build/issues/1675.

Mostly, the metacpan api url needed to be updated, but there were a few other fixes.

### CODE REVIEW

File: conda_build/api.py [hunk 1]
```diff
@@ -174,10 +174,11 @@ def skeletonize(packages, repo, output_dir=".", version=None, recursive=False,
     sources into expected conda recipe format."""
 
     version = getattr(config, "version", version)
+    #I dont now if these do anything
```
Comment (jerowe) [inline conda_build/api.py]: @msarahan , do these do anything? version, outdir, and recursive are not in kwargs.

Comment (msarahan) [inline conda_build/api.py]: I have no idea, I'm sorry.  I'm glad you put them there in main_skeleton.

File: conda_build/api.py [hunk 2]
```diff
@@ -174,10 +174,11 @@ def skeletonize(packages, repo, output_dir=".", version=None, recursive=False,
...
     if version:
         kwargs.update({'version': version})
     if recursive:
-        kwargs.update({'version': version})
+        kwargs.update({'recursive': recursive})
```
Comment (msarahan) [inline conda_build/api.py]: good catch.

**Comment by msarahan** [comment]
Looks like you need to catch a CondaRuntimeError for conda 4.2.x.  Otherwise fine.  Thanks!

Please sign the CLA, too, if you can.  If you need to consult your legal department, I can forward that info to the appropriate people here.

**Comment by jerowe** [comment]
@msaharn, not a problem, but I already signed a CLA with conda core.  Are they separate projects?

File: conda_build/api.py (hunk 1 above)
Comment (jerowe) [inline conda_build/api.py]: I'm reasonably certain they aren't doing anything, but I feel nervous about getting rid of things.

**Comment by msarahan** [comment]
@jerowe thanks - you only need to have signed one of the CLAs.

Eric is fat and happy most of the time.  Can't ask for more. =)  Thanks for asking.

**Comment by msarahan** [comment]
@jerowe fixed things up.  Sorry, they were a bit of a mess.  Needed to pull in CondaError from conda 4.2, not just 4.3.

This should pass CI now.  Will merge when it does.

**Comment by msarahan** [comment]
Hooray!  it worked!  merging.

**Comment by jerowe** [comment]
Thanks,  @msarahan!

**Pull request merged**